File: Same Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(p == null || q == null)
        {
            return p==null && q==null;
        }
        else
        {
            if(p.val == q.val)
            {
                return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
            }
            else
            {
                return false;
            }
        }
    }
}

File: Same Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode *p, TreeNode *q) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(p==NULL&&q==NULL)
        {
            return true;  
        }
        else if(p&&q)
        {
            if(p->val !=q->val)
            {
                return false;
            }
            else{
                bool left = isSameTree(p->left, q->left);
                if(left==false)
                    return false;
                else
                {
                    bool right = isSameTree(p->right, q->right);
                    return right&&left;
                }
            }
        }
        else
        {
            return false;
        }
    }
};